sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/SarifParserCallbackImpl.java

```diff
@@ -235,11 +232,6 @@ private static NewIssueLocation createIssueLocation(InputFile inputFile, Locatio
 
     } catch (IllegalArgumentException ex1) {
 
-      if (!isLocationResilient) {
-        // Our rules should fail if they report on an invalid location
-        throw ex1;
-      }
-
       try {
         // Precise location failed, now try the line...
         newIssueLocation = newIssueLocation.at(inputFile.selectLine(location.getStartLine()));
```

Comment:
It might be good to add some `debug` level log inside the catch clause. It might help investigation in the future if a user has issues reported not on a precise location